cmd/gen-go-sample/main.go

```diff
@@ -0,0 +1,227 @@
+	"flag"
+	"fmt"
+	"go/format"
+	"io/ioutil"
+	"log"
+	"os"
+	"sort"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/protoc-gen-go/descriptor"
+	"github.com/googleapis/gapic-generator-go/internal/errors"
+	"github.com/googleapis/gapic-generator-go/internal/pbinfo"
+	"github.com/googleapis/gapic-generator-go/internal/printer"
+	yaml "gopkg.in/yaml.v2"
+)
+
+func main() {
+	descFname := flag.String("desc", "", "proto descriptor")
+	gapicFname := flag.String("gapic", "", "gapic config")
+	nofmt := flag.Bool("nofmt", false, "skip gofmt, useful for debugging code with syntax error")
+	flag.Parse()
+
+	var (
+		gen = generator{
+			imps: map[pbinfo.ImportSpec]bool{},
+		}
+
+		donec = make(chan struct{})
+	)
+
+	go func() {
+		f, err := os.Open(*gapicFname)
+		if err != nil {
+			log.Fatal(errors.E(err, "cannot read GAPIC config file"))
+		}
+		defer f.Close()
+
+		if err := yaml.NewDecoder(f).Decode(&gen.gapic); err != nil {
+			log.Fatal(errors.E(err, "error reading GAPIC config file"))
+		}
+		donec <- struct{}{}
+	}()
+	go func() {
+		descBytes, err := ioutil.ReadFile(*descFname)
+		if err != nil {
+			log.Fatal(errors.E(err, "cannot read proto descriptor file"))
+		}
+
+		if err := proto.Unmarshal(descBytes, &gen.desc); err != nil {
+			log.Fatal(errors.E(err, "error reading proto descriptor file"))
+		}
+
+		gen.dinfo = pbinfo.Of(gen.desc.GetFile())
+		donec <- struct{}{}
+	}()
+
+	<-donec
+	<-donec
+
+	for _, iface := range gen.gapic.Interfaces {
+		for _, meth := range iface.Methods {
+			valSets := map[string]GapicValueSet{}
+			for _, vs := range meth.SampleValueSets {
+				valSets[vs.ID] = vs
+			}
+
+			for _, sam := range meth.Samples.Standalone {
+				for _, vsID := range sam.ValueSets {
+					vs, ok := valSets[vsID]
+					if !ok {
+						log.Fatal(errors.E(nil, "value set not found: %q", vsID))
+					}
+
+					gen.reset()
+					if err := gen.genSample(iface.Name, meth.Name, vs); err != nil {
+						log.Fatal(errors.E(err, "value set: %q", vs))
+					}
+					if err := gen.commit(!*nofmt); err != nil {
+						log.Fatal(errors.E(err, "value set: %q", vs))
+					}
+				}
+			}
+		}
+	}
+}
+
+type generator struct {
+	desc  descriptor.FileDescriptorSet
+	dinfo pbinfo.Info
+	gapic GapicConfig
+
+	pt   printer.P
+	imps map[pbinfo.ImportSpec]bool
+}
+
+func (g *generator) reset() {
+	g.pt.Reset()
+	for imp := range g.imps {
+		delete(g.imps, imp)
+	}
+}
+
+func (g *generator) commit(gofmt bool) error {
+	// We'll gofmt unless user asks us to not, so no need to think too hard about sorting
+	// "correctly". We just want a deterministic output here.
+	var imps []pbinfo.ImportSpec
+	for imp := range g.imps {
+		imps = append(imps, imp)
+	}
+	sort.Slice(imps, func(i, j int) bool {
+		if imps[i].Path != imps[j].Path {
+			return imps[i].Path < imps[j].Path
+		}
+		return imps[i].Name < imps[j].Name
+	})
+
+	var file bytes.Buffer
+	file.WriteString("package main\n")
+	file.WriteString("import(\n")
+	for _, imp := range imps {
```

Comment:
You'll need to import `fmt`, and have a line between that and non-stdlib imports

Reply:
Done. I thought gofmt would separate them; apparently it doesn't :(